apps/coordinator/src/components/Wallet/TransactionsTab/FeeBumping/components/RBF/RBFForm.tsx

```diff
@@ -0,0 +1,610 @@
+  const estimatedNewFee = useMemo(() => {
+    const txVsize = recommendation.vsize || 250; // Use default if not available
+    return Math.ceil(txVsize * selectedFeeRate).toString();
+  }, [selectedFeeRate, recommendation]);
+
+  // Update parent component
+  const updateParentWithDebounce = useCallback(
+    (newFeeLevel: string, newFeeRate: number) => {
+      // Clear any existing timeout to avoid multiple updates
+      if (updateTimeoutRef.current) {
+        clearTimeout(updateTimeoutRef.current);
+      }
+
+      // Mark that we're in the process of updating
+      isUpdatingFeeLevel.current = true;
+
+      // Set a debounced update
+      updateTimeoutRef.current = setTimeout(() => {
+        // For custom fee level, just update the fee rate
+        if (newFeeLevel === FEE_LEVELS.CUSTOM) {
+          onFeeRateChange(newFeeRate);
+        }
+        // For predefined levels, update priority first, then fee rate if needed ( else there is a conflict)
+        else if (feeLevels[newFeeLevel]) {
+          const calculatedFeeRate = Math.ceil(feeLevels[newFeeLevel]);
+
+          // Determine new priority
+          let newPriority: FeePriority = selectedPriority;
+          if (newFeeLevel === FEE_LEVELS.LOW) {
+            newPriority = FeePriority.LOW;
+          } else if (newFeeLevel === FEE_LEVELS.MEDIUM) {
+            newPriority = FeePriority.MEDIUM;
+          } else if (newFeeLevel === FEE_LEVELS.HIGH) {
+            newPriority = FeePriority.HIGH;
+          }
+
+          // Update priority if it changed and update callback exists
+          if (onPriorityChange && newPriority !== selectedPriority) {
+            onPriorityChange(newPriority);
+          }
+
+          // Update fee rate if different from current
+          if (calculatedFeeRate !== selectedFeeRate) {
+            onFeeRateChange(calculatedFeeRate);
+          }
+        }
+
+        // Reset updating flag
+        isUpdatingFeeLevel.current = false;
+        updateTimeoutRef.current = null;
+      }, 100); // 100ms debounce
+    },
+    [
+      feeLevels,
+      onFeeRateChange,
+      onPriorityChange,
+      selectedFeeRate,
+      selectedPriority,
+    ],
+  );
+
+  // Sync local customFeeRate with selectedFeeRate prop changes (coming from parent)
+  useEffect(() => {
+    // Don't update if we're in the middle of our own update process
+    if (!isUpdatingFeeLevel.current) {
+      setCustomFeeRate(selectedFeeRate);
+    }
+  }, [selectedFeeRate]);
+
+  // Set initial fee level based on priority
+  useEffect(() => {
+    if (isInitialRender.current && selectedPriority) {
+      setFeeLevel(PRIORITY_TO_FEE_LEVEL[selectedPriority]);
+      isInitialRender.current = false;
+    }
+  }, [selectedPriority]);
+
+  // Handle RBF type change
+  const handleRbfTypeChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setRbfType(event.target.value as "accelerate" | "cancel");
```

Comment:
you could create a type for the strings. I wouldn't be surprised if there was some way to pass as a generic to the event input type so this is inherited too